### DISCUSSION TITLE
I'd like some 'retry' functionality built into Dark

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4376] [Suggestions]
(I thought about this while working on the new HttpClient)

What does 'retry' look like in Dark long-term?

**Comment by StachuDotNet** [comment]
If this were handled in stdlib functions alone, here's a really rough sketch of a function I'd reach for:
```
let retryIfError
    (fn: params -> Result<A,B>)
    (retryCount: int)
  = ...
```

Likely, we'd also have a lower-level `retry` function, like
```
let retry
    (fn: params -> A)
    (passes: A -> bool)
    (retryCount: int)
  = ...
```

**Comment by StachuDotNet** [comment]
There's probably room for something much more sophisticated than a simple standard library function, though. I imagine much fancier 'retry' buttons in the Editor - an extension of the current analysis functionality.

**Comment by pbiggar** [comment]
Can you describe the problem? You want to retry HttpClient calls?

**Reply by StachuDotNet** [reply]
Actually, I think I was confusing myself, conflating the current redirect behavior (which loops up to X times - 50 I think) with an imaginary 'retry' behaviour.

I don't have any specific need to 'retry' httpclient calls, or retry other things - I'll close this, and someone can open up a more sophisticated discussion down the road.
